detector/aws_ecs_cluster_duplicate_name_test.go

```diff
@@ -0,0 +1,138 @@
+package detector
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/golang/mock/gomock"
+	"github.com/k0kubun/pp"
+	"github.com/wata727/tflint/config"
+	"github.com/wata727/tflint/issue"
+	"github.com/wata727/tflint/mock"
+)
+
+func TestDetectAwsECSDuplicateName(t *testing.T) {
+	cases := []struct {
+		Name     string
+		Src      string
+		State    string
+		Response []*ecs.Cluster
+		Issues   []*issue.Issue
+	}{
+		{
+			Name: "Cluster name is duplicate",
+			Src: `
+resource "aws_ecs_cluster" "foo" {
+  name = "white-hart"
+}`,
+			Response: []*ecs.Cluster{
+				{
+					ClusterName: aws.String("white-hart"),
+				},
+				{
+					ClusterName: aws.String("black-hart"),
+				},
+			},
+			Issues: []*issue.Issue{
+				{
+					Detector: "aws_ecs_cluster_duplicate_name",
+					Type:     "ERROR",
+					Message:  "\"white-hart\" is duplicate name. It must be unique.",
+					Line:     3,
+					File:     "test.tf",
+				},
+			},
+		},
+		{
+			Name: "Cluster name is unique",
+			Src: `
+resource "aws_ecs_cluster" "foo" {
+  name = "white-hart"
+}`,
+			Response: []*ecs.Cluster{
+				{
+					ClusterName: aws.String("brown-hart"),
+				},
+				{
+					ClusterName: aws.String("black-hart"),
+				},
+			},
+			Issues: []*issue.Issue{},
+		},
+		{
+			Name: "Cluster is duplicate, but exists in state",
+			Src: `
+resource "aws_ecs_cluster" "foo" {
+  name = "white-hart"
+}`,
+			State: `
+{
+    "modules": [
+        {
+            "resources": {
+                "aws_ecs_cluster.foo": {
+                    "type": "aws_ecs_cluster",
+                    "depends_on": [],
+                    "primary": {
+                        "id": "arn:aws:ecs:us-east-1:hogehoge:cluster/white-hart",
+                        "attributes": {
+                            "id": "arn:aws:ecs:us-east-1:hogehoge:cluster/white-hart",
+                            "name": "white-hart"
+                        },
+                        "meta": {},
+                        "tainted": false
+                    },
+                    "deposed": [],
+                    "provider": ""
+                }
+            }
+        }
+    ]
+}
+`,
+			Response: []*ecs.Cluster{
+				{
+					ClusterName: aws.String("white-hart"),
+				},
+				{
+					ClusterName: aws.String("black-hart"),
+				},
+			},
+			Issues: []*issue.Issue{},
+		},
+	}
+
+	for _, tc := range cases {
+		c := config.Init()
+		c.DeepCheck = true
+
+		awsClient := c.NewAwsClient()
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+		ecsmock := mock.NewMockECSAPI(ctrl)
```

Comment:
__[gometalinter]__
* [staticcheck] NewMockECSAPI not declared by package mock
* [gosimple] NewMockECSAPI not declared by package mock
* [varcheck] unused variable or constant NewMockECSAPI not declared by package mock

<!-- Comment by [SideCI](https://sideci.com) -->